core/src/com/group66/game/cannon/BallGraphAdjacentIterator.java

```diff
@@ -57,5 +57,13 @@ public Ball next() {
         list.remove(0);
         return ret;
     }
+    
+    /**
+     * remove is not supported by the iterator
+     */
+    @Override
+    public void remove() {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
In java 1.8 the function in the interfaced is changed in a default operation, so you don't have to override it. In java 1.7 it is not.